spec/documents-spec.js

```diff
@@ -0,0 +1,270 @@
+    .set('x-access-token', token)
+    .end((err, res) => {
+      expect(res.status).toBe(404);
+      expect(res.body.error).toBe('Could not find document.');
+      done();
+    });
+  });
+
+  it('Updates documents', (done) => {
+    request
+    .put(`/api/documents/${documentId}`)
+    .set('x-access-token', token)
+    .send({
+      content: 'Hello',
+    })
+    .end((err, res) => {
+      expect(res.status).toBe(200);
+      expect(res.body.message).toBe('Document updated successfully.');
+      done();
+    });
+  });
+
+  it('Rejects duplicate titles', (done) => {
+    request
+    .put(`/api/documents/${documentId}`)
+    .set('x-access-token', token)
+    .send({
+      title: 'Lord of the Rings',
+    })
+    .end((err, res) => {
+      expect(res.status).toBe(409);
+      expect(res.body.error).toBe('Duplicate entry.');
+      done();
+    });
+  });
+
+  it('Does not update non-existant documents', (done) => {
+    request
+    .put('/api/documents/57cd551476d7f2790eb87c47')
+    .set('x-access-token', token)
+    .send({
+      title: 'Lord of the Rings',
+    })
+    .end((err, res) => {
+      expect(res.status).toBe(404);
+      expect(res.body.error).toBe('Document not found.');
+      done();
+    });
+  });
+
+  it('Does not update documents if no data is provided', (done) => {
+    request
+    .put(`/api/documents/${documentId}`)
+    .set('x-access-token', token)
+    .send({})
+    .end((err, res) => {
+      expect(res.status).toBe(400);
+      expect(res.body.error).toBe('Nothing to update.');
+      done();
+    });
+  });
+
+  it('Does not update documents not owned by logged in user', (done) => {
+    request
+    .put('/api/documents/57c975eb2c3d08864b51cd0a')
+    .set('x-access-token', token)
+    .send({
+      title: 'Potter Head',
+    })
+    .end((err, res) => {
+      expect(res.status).toBe(403);
+      expect(res.body.error).toBe('Cannot edit document you did not create.');
+      done();
+    });
+  });
+
+  it('Deletes a document by id', (done) => {
+    request
+    .delete(`/api/documents/${documentId}`)
+    .set('x-access-token', token)
+    .end((err, res) => {
+      expect(res.status).toBe(200);
+      expect(res.body.message).toBe('Document deleted successfully.');
+      done();
+    });
+  });
+
+  it('Does not delete non-existant documents', (done) => {
+    request
+    .delete('/api/documents/sfr456jgih6hv39y5343')
+    .set('x-access-token', token)
+    .end((err, res) => {
+      expect(res.status).toBe(404);
+      expect(res.body.error).toBe('Document not found.');
+      done();
+    });
+  });
+
+  it('Does not delete documents not owned by logged in user', (done) => {
+    request
+    .delete('/api/documents/57c975eb2c3d08864b51cd0a')
+    .set('x-access-token', token)
+    .end((err, res) => {
+      expect(res.status).toBe(403);
+      expect(res.body.error).toBe('Cannot delete document you did not create.');
+      done();
+    });
+  });
+});
```

Comment:
For all your tests, it is a good idea to check whether the returned data is in the expected format. i.e Object, array e.t.c